.github/CONTRIBUTING.md

```diff
@@ -212,7 +212,21 @@ invoke `make format`.
 The target `make formatall` in the root will automatically run all style checks
 and make any required changes. PRs must pass the formatting checks before landing.
 
-### Step 8: Landing
+### Step 8: Licensing
+
+Tock is dual-licensed under Apache2/MIT. Files in Tock require a short header
+indicating this. In 99% of cases, the following text should be at the top of every file:
+
+> Copyright The Tock Project Authors.
+>
+> Dual licensed under the Apache License, Version 2.0 or the MIT License, at your option.
+> See the License for the specific language governing permissions and limitations under the License.
+> This file may not be copied, modified, or distributed except according to those terms.
+
+This should be automatically enforced by CI tooling. If you need different licensure
+for some reason, please get in touch with the core team directly.
+
```

Comment:
IANAL, but I don't think the way Tock contributions are setup right now, it would be possible to re-licence Tock in a different licence unless all contributors agree. Re-licensing would typically be possible when there has been copyright assignment. I don't think neither ASL nor MIT requires copyright assignment.

Reply:
Indeed. I rephrased that slightly. The goal is basically to try to be friendly still with this text :)

Reply:
IAMAL, but I'm also wondering what `we will see if we can work out a solution` would entail. I see the following possible outcomes:
1. After all, the contributor agrees to Apache License, Version 2.0 + MIT License. Easy case.
1. Another license is proposed. Accepting the contribution would likely require re-licensing the whole of Tock, which would mean either of:
  a. Agreement from all the existing [contributors](https://github.com/tock/tock/graphs/contributors) (currently 94), as well as their (former) employers in case such employers hold copyright, their heirs in case of a tragic event, etc. This is basically intractable.
  b. Removing commits from existing contributors who don't agree on the re-licensing, or don't reply in time. Depending on the scale of the contribution, this can affect core functionality of Tock.
1. The contributor creates a fork of the project, with their contributions in this fork licensed however they like as long as it complies with either of Apache License, Version 2.0 or MIT License, at their option. But forking isn't really "contributing" in the scope of this page.

All in all, (1) is a non-event (apart from interesting discussions about licensing & copyright), (2) is a tremendous amount of work (even if the contribution is good, this is likely not worth it) and (3) is not "contributing" per-se.

Also, I'm not sure there will even be such a case of a contributor coming to Tock but requesting another license. Apache + MIT are a common setup for open-source nowadays, and anyone seeing the project is aware of the license before they decide to contribute or not.

Reply:
Okay then, I just removed that sentence.